pylint_odoo/misc.py

```diff
@@ -172,6 +166,55 @@ def _check_{NAME_KEY}(self, module_path)
                         node.file = node_file_original
                         node.lineno = node_lineno_original
 
+    def visit_module(self, node):
+        self.wrapper_visit_module(node)
+
+    def add_message(self, *args, **kwargs):
+        version = (self.manifest_dict.get('version')
+                   if isinstance(self.manifest_dict, dict) else '')
+        match = re.match(r"(?P<version>\d+.\d+)\.\d+\.\d+\.\d+$", version)
+        name_check = [arg for arg in args]
+        if match and len(name_check) >= 1:
+            if not self._is_version_supported(match.group('version'),
+                                              name_check[0]):
+                return
+        return super(PylintOdooChecker, self).add_message(*args, **kwargs)
+
+    def _is_version_supported(self, version, name_check):
+        if not hasattr(self, 'odoo_check_versions'):
+            return True
+        odoo_check_versions = self.odoo_check_versions.get(name_check, {})
+        if not odoo_check_versions:
+            return True
+        version = LooseVersion(version)
+        min_odoo_version = LooseVersion(odoo_check_versions.get(
+            'min_odoo_version', DFTL_VALID_ODOO_VERSIONS[0]))
+        max_odoo_version = LooseVersion(odoo_check_versions.get(
+            'max_odoo_version', DFTL_VALID_ODOO_VERSIONS[-1]))
+        return (version >= min_odoo_version and version <= max_odoo_version)
```

Comment:
What about use a `between` style python: `min_odoo_version <= version <= max_odoo_version`?

Reply:
Done, thanks for the trick about the python between